main.py

```diff
@@ -1,70 +1,92 @@
+    def date_is_datetime(cls, value):
+        """Verify that date is datetime"""
+        try:
+            sales_date = datetime.strptime(value, "%Y-%m-%d").date()
+            return sales_date
+        except ValueError as value_error:
+            raise HTTPException(
+                status_code=400,
+                detail="Invalid sales date format. Use the YYYY-MM-DD format.",
+            ) from value_error
 
-@app.post("/api/sales")
-async def create_sale(sale: Sale):
+    @validator("buyer")
+    def name_is_letters_and_spaces(cls, value):
+        """Verify that name is letters and spaces"""
+        if not all(x.isalpha() or x.isspace() for x in value):
+            raise HTTPException(
+                status_code=400,
+                detail="Invalid sales person name. Name cannot contain numbers.",
+            )
+        return value
 
-    # Validate the input data
-    try:
-        sales_date = datetime.strptime(sale.sales_date, "%Y-%m-%d").date()
-    except ValueError:
-        raise HTTPException(
-            status_code=400,
-            detail="Invalid sales date format. Use the YYYY-MM-DD format.",
-        )
+    @validator("apples")
+    def apples_is_int(cls, value):
+        """Verify that number of apples is int"""
+        try:
+            return int(value)
+        except ValueError as value_error:
+            raise HTTPException(
+                status_code=400,
+                detail="Apples must be an integer",
+            ) from value_error
 
-    if not sale.buyer.isalpha():
-        raise HTTPException(
-            status_code=400,
-            detail="Invalid sales person name. Name cannot contain numbers.",
-        )
+    @validator("oranges")
+    def oranges_is_int(cls, value):
+        """Verify that number of oranges is int"""
+        try:
+            return int(value)
+        except ValueError as value_error:
+            raise HTTPException(
+                status_code=400,
+                detail="Oranges must be an integer",
+            ) from value_error
 
-    if not sale.apples:
-        sale.apples = 0
-    if not sale.oranges:
-        sale.oranges = 0
 
-    # Insert the data into the shopping_list table
+@app.post("/api/sales")
+async def create_sale(sale: Sale):
+    """Function to insert the data into the sales table"""
     cursor = cnxn.cursor()
+    create_db = f"IF NOT EXISTS(SELECT * FROM sys.databases WHERE name = '{DB_NAME}')\
+               BEGIN CREATE DATABASE [{DB_NAME}] END"
+    cursor.execute(create_db)
+    change_db = f"USE {DB_NAME}"
+    cursor.execute(change_db)
+
+    create_table = (
+        "IF NOT EXISTS(SELECT name FROM sys.tables WHERE name = 'Inventory') BEGIN;"
+    )
+    table_data = "CREATE TABLE Inventory (sales_id INT NOT NULL IDENTITY PRIMARY KEY, \
+                  sales_date VARCHAR(10), buyer VARCHAR(30),  apples INT, oranges INT);"
+    end_table = "END"
+    cursor.execute(create_table + table_data + end_table)
```

Comment:
The logic for managing a table and inserting into it could be a separate function.